### PR TITLE
feat(reportImport): add ReportImport agent

### PR DESCRIPTION
**Pull request by maxhbr**
**NOTE: this is work in progress and this PR should be here as place to aggregate discussion related to the feature**

corresponding issue: #669
related PR: https://github.com/maxhbr/fossology/pull/2

### CODE REVIEW

**Comment by maxhbr** [comment]
Currently the UI look like:

**Comment by maxhbr** [comment]
NOTE: This PR currently also includes the changes from https://github.com/fossology/fossology/pull/889.

**Comment by shaheemazmalmmd** [comment]
Can you please add this new agent to debian packages ??

**Comment by maxhbr** [comment]
@shaheemazmalmmd: I have made the corresponding changes (they are untested)

**Comment by shaheemazmalmmd** [comment]
@maxhbr : There is a error while building a package. 
`dpkg-gencontrol: error: illegal package name 'fossology-reportImport': character 'I' not allowed
dh_gencontrol: dpkg-gencontrol -pfossology-reportImport -ldebian/changelog -Tdebian/fossology-reportImport.substvars -Pdebian/fossology-reportImport returned exit code 255"`

Can you please have a look into this

**Comment by maxhbr** [comment]
@shaheemazmalmmd the "illegal char"-problem should be fixed

**Comment by maxhbr** [comment]
@shaheemazmalmmd The code here has currently a bug which prevents the frontend arguments from beeing passed to the backend.

**Comment by maxhbr** [comment]
@shaheemazmalmmd @mcjaeger this branch is fixed again and should work

**Comment by maxhbr** [comment]
@mcjaeger this PR is now based on https://travis-ci.org/fossology/fossology/jobs/279648074 and imports copyrights as textfindings

**Comment by maxhbr** [comment]
The travis tests are ran in https://github.com/maxhbr/fossology/pull/2

**Comment by shaheemazmalmmd** [comment]
@maxhbr : Please fix the notices  

[Fri Sep 29 16:52:17.075983 2017] [:error] [pid 22009] [client 10.0.2.2:59549] PHP Notice:  Undefined offset: 0 in /usr/local/share/fossology/lib/php/common-agents.php on line 312, referer: http://localhost:3311/repo/?mod=agent_add
[Fri Sep 29 16:52:17.102987 2017] [:error] [pid 22009] [client 10.0.2.2:59549] PHP Notice:  Undefined offset: 0 in /usr/local/share/fossology/lib/php/common-agents.php on line 312, referer: http://localhost:3311/repo/?mod=agent_add

**Comment by shaheemazmalmmd** [comment]
Not able to view the **copyrights** blank page appears 
To reproduce:

Download SPDX RDF from `Master` for package X

In a clean instance of `dev/addSPDX2ImportAgent_dist` :
Upload a package and do not run any agents.
Go to reportImport upload 'SPDX RDF X' 
Then Click on ' Copyright/Email/Url ' in browse page



``PHP Warning:  pg_prepare(): Query failed: ERROR:  relation &quot;copyright_ars&quot; does not exist\nLINE 1: SELECT * FROM (SELECT agent_pk, agent_name FROM copyright_ar...\n                                                        ^ in /usr/local/share/fossology/lib/php/Db/Driver/Postgres.php on line 62, referer: http://localhost:3311/repo/?mod=view-license&upload=2&item=51
[2017-10-03 13:13:40] default.CRITICAL: ERROR:  relation "copyright_ars" does not exist LINE 1: SELECT * FROM (SELECT agent_pk, agent_name FROM copyright_ar...                                                         ^ [] [], referer: http://localhost:3311/repo/?mod=view-license&upload=2&item=51
PHP Fatal error:  Uncaught Fossology\\Lib\\Exception: error executing: SELECT * FROM (SELECT agent_pk, agent_name FROM copyright_ars, agent\n        WHERE agent_fk=agent_pk AND upload_fk=2 AND agent_enabled ORDER BY agent_fk DESC limit 1) latest_copyright -- Fossology\\Lib\\Proxy\\LatestScannerProxy::getNameToIdMap.latest_scanner2_copyright\n\nERROR:  relation "copyright_ars" does not exist\nLINE 1: SELECT * FROM (SELECT agent_pk, agent_name FROM copyright_ar...\n                                                        ^ in /usr/local/share/fossology/lib/php/Db/DbManager.php:141\nStack trace:\n#0 /usr/local/share/fossology/lib/php/Db/ModernDbManager.php(51): Fossology\\Lib\\Db\\DbManager->checkResult(false, 'SELECT * FROM (...')\n#1 /usr/local/share/fossology/lib/php/Proxy/LatestScannerProxy.php(89): Fossology\\Lib\\Db\\ModernDbManager->prepare('Fossology\\\\Lib\\\\P...', 'SELECT * FROM (...')\n#2 /usr/local/share/fossology/lib/php/Dao/AgentDao.php(210): Fossology\\Lib\\Proxy\\LatestScannerProxy->getNameToIdMap()\n#3 /usr/local/share/fossology/copyright/u in /usr/local/share/fossology/lib/php/Db/DbManager.php on line 141, referer: http://localhost:3311/repo/?mod=view-license&upload=2&item=51 ``

**Comment by maxhbr** [comment]
The problem might be that the copyright table was not yet created. Maybe upload another file and run copyright for it.

**Comment by shaheemazmalmmd** [comment]
Sure, But copyright table gets created when we do postinstall, only Ecc table depends on the agent code.

**Comment by shaheemazmalmmd** [comment]
The problem is with "copyright_ars" it gets created only through agent it seems.

**Comment by maxhbr** [comment]
Yes, you are right, `copyright_ars` is missing if no copyright was ran.

**Comment by shaheemazmalmmd** [comment]
For license decisions only files with "identified" are added as decisions. other decisions are empty. 

Please Check no radio button selected.

itshows3highlightstringsbutnothingworks

**Comment by mcjaeger** [comment]
Hi, I think the point is true (given the links is displayed) let's skip the highlight, as this is not supported by the SPDX import format. regarding the radio buttono selection, it should be selected if there is a import and this import was set to already set decisions, this is the point?

File: src/reportImport/ui/Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright Siemens AG 2014
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply src/reportImport/ui/Makefile]: This file was not changed enough to be worth updating the copyright year. It is just copied from one of the other agents.

File: src/reportImport/agent/services.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+/*
+ Copyright Siemens AG 2015
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply src/reportImport/agent/services.php]: This file is just a copy and thus should keep its header without changes

File: src/reportImport/agent/XmlImportSource.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      $allFiles[$fileName] = $fileName;
+    }
+    return $allFiles;
+  }
+
+  /**
+   * @param $fileid
+   * @return array
+   */
+  public function getHashesMap($fileid)
```
[the remark this answers is not in the document]
Comment (maxhbr) [reply src/reportImport/agent/XmlImportSource.php]: That is specified in the interface `ImportSource`

**Comment by shaheemazmalmmd** [comment]
@maxhbr : Seems like there is some issue with existing database on this branch. 

**To reproduce** 

-> install master branch upload 2 components to fossology and do some clearing.
-> Now install report import branch upload a component. (In my case i have uploaded zlib).

reportimport-1

-> Click on upload and import.
-> It schedules report import for previously uploaded component. Not the one which i have selected.

reportimport2

File: src/reportImport/ui/ReportImportPlugin.php [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    if (empty($uploadId) ||
+        !array_key_exists('report',$_FILES) ||
+        sizeof($_FILES['report']['name']) != 1)
+    {
+      return $this->showUiToChoose($uploadId);
+    }
+    else
+    {
+
+      $jobMetaData = $this->runImport($uploadId, $_FILES['report'], $request);
```
Comment (shaheemazmalmmd) [inline src/reportImport/ui/ReportImportPlugin.php]: Here the ``$uploadId`` will be always the value of  first upload deom the hidden field.  Adding this line `` $uploadId = intval(GetArrayVal("uploadselect", $_POST));`` in else case will give the value of selected upload.

**Comment by maxhbr** [comment]
@shaheemazmalmmd The mentioned bug is fixed and the value of `"uploadselect"` is used. But there is still the bug that one is redirected to the job-view of the wrong upload (where no job was scheduled).

**Comment by mcjaeger** [comment]
just for the files: I did a retest this morning using the vagrant deployment with the latest commit, everything looks fine.

**Pull request merged**